product_listings_manager/app.py

```diff
@@ -17,5 +17,12 @@
 app.register_blueprint(rest_api_v1.blueprint, url_prefix='/api/v1.0')
 xmlrpc.handler.connect(app, '/xmlrpc')
 
+
+@app.errorhandler(404)
+def handle_api_error(ex):
```

Comment:
This error is specific for 404, so maybe the method name change to page_not_found?

Reply:
Good idea. I've renamed it to `page_not_found_error`